third_party/move/move-model/src/ty.rs

```diff
@@ -979,17 +979,17 @@ impl Type {
 
     /// Returns true if this is any number type.
     pub fn is_number(&self) -> bool {
-        if let Type::Primitive(p) = self {
-            if let PrimitiveType::U8
+        if let Type::Primitive(
+            PrimitiveType::U8
             | PrimitiveType::U16
             | PrimitiveType::U32
             | PrimitiveType::U64
             | PrimitiveType::U128
             | PrimitiveType::U256
-            | PrimitiveType::Num = p
-            {
-                return true;
-            }
+            | PrimitiveType::Num,
+        ) = self
+        {
+            return true;
```

Comment:
hmm... this could probably be a `matches!`, but seems acceptable.

Interesting that it accepted return true, after other places did not.